app.json

```diff
@@ -0,0 +1,137 @@
+    },
+    "EMAIL_PORT": {
+      "description": "SMTP server port",
+      "value": "465",
+      "required": false
+    },
+    "EMAIL_HOST_USER": {
+      "description": "SMTP server username",
+      "value": "@gmail.com",
+      "required": false
+    },
+    "EMAIL_HOST_PASSWORD": {
+      "description": "SMTP server password",
+      "value": "",
+      "required": false
+    },
+    "SERVER_EMAIL": {
+      "description": "Sender email address",
+      "value": "<shynet> noreply@shynet.example.com",
+      "required": false
+    },
+    "DJANGO_SECRET_KEY": {
+      "description": "General Django settings",
+      "generator": "secret"
+    },
+    "ALLOWED_HOSTS": {
+      "description": "For better security, set this to your deployment's domain. Comma separated.",
+      "value": "*",
+      "required": false
+    },
+    "SIGNUPS_ENABLED": {
+      "description": "Set to True (capitalized) if you want people to be able to sign up for your Shynet instance (not recommended)",
+      "value": "False",
+      "required": false
+    },
+    "TIME_ZONE": {
+      "description": "The timezone of the admin panel. Affects how dates are displayed.",
+      "value": "America/New_York",
+      "required": false
+    },
+    "SCRIPT_USE_HTTPS": {
+      "description": "Set to 'False' if you will not be serving content over HTTPS",
+      "value": "True",
+      "required": false
+    },
+    "SCRIPT_HEARTBEAT_FREQUENCY": {
+      "description": "How frequently should the monitoring script 'phone home' (in ms)?",
+      "value": "5000",
+      "required": false
+    },
+    "SESSION_MEMORY_TIMEOUT": {
+      "description": "How much time can elapse between requests from the same user before a new session is created, in seconds?",
+      "value": "1800",
+      "required": false
+    },
+    "ONLY_SUPERUSERS_CREATE": {
+      "description": "when you'd like to invite others to your Shynet instance but don't want them to be able to create services of their owShould only superusers (admins) be able to create services? This is helpfuln.",
+      "value": "True",
+      "required": false
+    },
+    "PERFORM_CHECKS_AND_SETUP": {
+      "description": "Whether to perform checks and setup at startup, including applying unapplied migrations. For most setups, the recommended value is True. Defaults to True. Will skip only if value is False.",
+      "value": "True",
+      "required": false
+    },
+    "SHYNET_ADMIN_EMAIL": {
+      "description": "Your admin user email. A temporary password will be printed to the console on first run.",
+      "value": "admin@example.com",
+      "required": false
+    },
+    "SHYNET_HOST": {
+      "description": "The domain on which you'll be hosting Shynet.",
+      "value": "your-app.herokuapp.com",
+      "required": false
+    },
+    "SHYNET_WHITELABEL": {
+      "description": "What you'd like to call your Shynet instance.",
+      "value": "shynet-analytics",
+      "required": false
+    }
```

Comment:
These settings were removed in Shynet v0.5.0; the updated usage guide uses separate commands to set these options.

The reason I removed these variables is that they were only used on the first run during initialization of the database; people rightfully got confused when they tried to change their values but nothing happened. Moving the initialization process to use _commands_ better matches peoples' mental model of how these will behave.

We need some other way to setup the host, whitelabel, and admin account. Is there a way to tell Heroku to run some commands when this is being initially created?

Reply:
> These settings were removed in Shynet v0.5.0; the updated usage guide uses separate commands to set these options.
> 

It was realy removed? I found references of {`SHYNET_ADMIN_EMAIL`, `SHYNET_WHITELABEL` and `SHYNET_HOST`} on {`GUIDE.md`, `startup_checks.sh`} and I thought they were still being used.


> The reason I removed these variables is that they were only used on the first run during initialization of the database; people rightfully got confused when they tried to change their values but nothing happened. Moving the initialization process to use _commands_ better matches peoples' mental model of how these will behave.
> 
I think this can be handle on "create admin user upon first boot on initialization" feature because even if I already got a database to restore, I may want to be able to use this environment variables to make sure I always have the same user/credentials.
Since environment variables are handle for your own infrastructure valt solution don't see any secure issues.


> We need some other way to setup the host, whitelabel, and admin account. Is there a way to tell Heroku to run some commands when this is being initially created?
>

For that I usually solve this in the `postinstall` script (only runs when `NODE_ENV == dev`) of package.json in a nodeJS project. I dont realy know how phyton stack handle fit this idea.
Somehow, currently this PR deploy shynet on heroku by running it as a container. It means Heroku only follow entrypoint defined on [Dockerfile](https://github.com/milesmcc/shynet/blob/master/Dockerfile#L35).

I think this gives the ability to deploy multiple environments managed {on,by} heroku.

Note that solving this problem on Dockerfile give us a more vendor independent solution.

Reply:
Just checked, these _have_ been removed. Maybe you started this PR before v0.5.0, but fortunately the current setup guide doesn't mention these variables. They also aren't present in startup checks.

(It may be helpful to rebase your PR on the newest version of Shynet.)

Does Heroku offer some way to specify a script to run on first deploy? Or some commands to run? It would be nice if people could just put in the email address, whitelabel, and hostname and then have Heroku run the necessary setup commands for them.

Reply:
Heroku uses the entrypoint from the [container itself](https://github.com/milesmcc/shynet/blob/master/Dockerfile#L35). 

For this I [was required to  set `SHYNET_HOST` on `startup_checks.sh`](https://github.com/milesmcc/shynet/blob/1b344fb90cfb6b9e38cabc101ad1067ec9d736d9/shynet/startup_checks.sh#L18) so that [avoid this warning](https://github.com/milesmcc/shynet/blob/1b344fb90cfb6b9e38cabc101ad1067ec9d736d9/shynet/startup_checks.sh#L19) on boot.

What do you have in mind?

This way `startup_checks.sh` started each boot successfully and migrating database only on firt boot. 👍 

By removing {DB_NAME, DB_USER, DB_PASSWORD, DB_HOST} from `app.json` we wont be able to use with a different database then "addon heroku" itselfs.
For that reason I kept `SHYNET_HOST` in` app.json`.


After forked the main repo, clone my forked version and I started my work within a new branch `features/heroku-quick-deploy` from `origin/master`.

Commit SHA:
`1b344fb90cfb6b9e38cabc101ad1067ec9d736d9`

The same commit are the last on `origin/dev`, `origin/master` and `v0.5.0` tag.